authentication/api/certificate_extensions.py

```diff
@@ -0,0 +1,20 @@
+from asn1crypto.core import SequenceOf, UTF8String
+
+
+class Roles(SequenceOf):
+    _child_spec = UTF8String
+
+
+def decode(der_bytes: bytes):
```

Comment:
Slight quibble with the names here. `decode_roles()` would be better for this function, because there is also a Application extension.

Similar for `encode()`